### PR TITLE
making vn2 use a config file

### PR DESCRIPTION
**Pull request by SethHollandsworth**
Currently there are lots of configuration arguments to be used for VN2. This improves the user experience by being able to check in configuration with git to give more reproducible operations.

### CODE REVIEW

**Review by hgarvison — APPROVED** [review]

File: src/confcom/azext_confcom/template_util.py [hunk 1]
```diff
@@ -55,13 +55,17 @@ def case_insensitive_dict_get(dictionary, search_key) -> Any:
...
 def deep_dict_update(source: dict, destination: dict):
     """
     https://stackoverflow.com/questions/20656135/python-deep-merge-dictionary-data
     """
     for key, value in source.items():
         if isinstance(value, dict):
             node = destination.setdefault(key, {})
+            if node is None:
+                destination[key] = {}
+                node = destination
```
Comment (ksayid) [inline src/confcom/azext_confcom/template_util.py:68]: In L68, should node be the new empty dict?

File: src/confcom/azext_confcom/_params.py [hunk 2]
```diff
@@ -134,80 +134,14 @@ def load_arguments(self, _):
...
-            required=False,
-            help="KUBERNETES_PORT_443_TCP_PORT environment variable (default '443')",
-        )
-        c.argument(
-            "output_file_name",
-            options_list=("--output-file-name"),
-            required=False,
-            help="Name of the output file (default 'arm-template.json')",
-        )
-        c.argument(
```
Comment (ksayid) [inline src/confcom/azext_confcom/_params.py:198]: what do you think of moving --print-json out of the config? 
I feel like it's not really a configuration thing; it's more of a debug thing.

File: src/confcom/azext_confcom/template_util.py (hunk 1 above)
Comment (SethHollandsworth) [reply src/confcom/azext_confcom/template_util.py:68]: it should, I think I was hitting a case where the value in `destination` was `None`. I'll look a little more closely at the issue. I don't think this is the best place for that error check

**Review by ksayid — APPROVED** [review]
LGTM

**Pull request merged**